presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java

```diff
@@ -60,17 +62,27 @@ public Optional<RecordCursor> createRecordCursor(
         // make sure the FileSystem is created with the proper Configuration object
         Path path = new Path(fileSplit.getPath());
         try {
+            if (!fileSplit.getCustomSplitInfo().isEmpty() && isHudiRealtimeSplit(fileSplit.getCustomSplitInfo())) {
```

Comment:
I guess with realTimeSplits, customSplitInfo will never be empty. WDYT?

Reply:
Yes I believe that should be the case. But since this class is generic, there might be other formats that could go through this code flow as well in the future. That's the reason I have added this check.

Reply:
I feel checking if the customSplitInfo is empty is redundant, but I will leave the final decision to you.